pyogrio/raw.py

```diff
@@ -385,8 +391,10 @@ def write(
     metadata=None,
     dataset_options=None,
     layer_options=None,
+    timezone_cols_metadata=None,
     **kwargs,
 ):
+    kwargs.pop("dtypes", None)
```

Comment:
This is needed for our roundtrip tests because `read` now returns dtypes as well?

Reply:
Yep that's right- this was a quick hack to get some feedback on the tests without changing to much. When this is finalised the intent was to clean this up by doing the equivalent to this in individual tests, rather than pollute this function, I think that's cleaner?

Reply:
I had another look at this, we can move this pop to conftest.py::prepare_testfile for some cases, but there are still 22 tests that would need to pop the dtypes key. Another (still messy) option might be to add a kwarg to `read` along the lines of "include_dtypes_in_meta", defaulting to False, and is set to true in the datetime as string path (it could also be tied to the value of `datetime_as_string` but that feels confusing)

Reply:
cc @brendan-ward (in case you can think of a better solution)  

Context is that this PR adds a new entry to the `meta` dictionary returned by `pyogrio.raw.read`, i.e. a "dtypes" key to know what the actual dtype is according to the OGR fields (which we need to know that the field was datetime64). 
But in `test_raw_io.py`, we have lots of places where we do a read and then write, and pass the `meta` from read to the write unpacked as keyword arguments with `**meta`. 

However, this "dtypes" key in `meta` is not an argument of `pyogrio.raw.write`, in contrast with the other keys (crs/encoding/fields/geometry_type), and so that would error if we don't remove "dtypes" from `meta`.

So in the current PR, to not have the current "hack" above of removing `dtypes` from the kwargs _inside_ `write`, we would have to constantly do `meta.pop("dtypes")` before passing them as `**meta` in the tests.

Reply:
I quite like having `dtypes` returned in `meta` in general.  I think popping it here makes sense, just add a comment indicating that we are doing it explicitly to allow passing `meta` produced by `read` directly into `write`.